From 05769d831a48967bc5b71470ff7a1e7d58a6f35b Mon Sep 17 00:00:00 2001
From: Gustavo Solaira <gustavos@codeaurora.org>
Date: Wed, 13 Dec 2017 13:38:48 -0800
Subject: [PATCH] esoc: Disable access check for ESOC IOCTLs

Disable the access check on the ESOC IOCTLs so they
can be accessed by user space applications directly.

Change-Id: If3c3fff06b471744e9eb41aee496bf8cd06fe7f2
Signed-off-by: Gustavo Solaira <gustavos@codeaurora.org>
---
 drivers/esoc/esoc_dev.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/esoc/esoc_dev.c b/drivers/esoc/esoc_dev.c
index a1e7a52..df8688d 100644
--- a/drivers/esoc/esoc_dev.c
+++ b/drivers/esoc/esoc_dev.c
@@ -196,13 +196,9 @@ static long esoc_dev_ioctl(struct file *file, unsigned int cmd,
 		uhandle->cmd_eng_reg = true;
 		break;
 	case ESOC_CMD_EXE:
-		if (esoc_clink->cmd_eng != &uhandle->eng)
-			return -EACCES;
 		get_user(esoc_cmd, (u32 __user *)arg);
 		return clink_ops->cmd_exe(esoc_cmd, esoc_clink);
 	case ESOC_WAIT_FOR_REQ:
-		if (esoc_clink->req_eng != &uhandle->eng)
-			return -EACCES;
 		err = wait_event_interruptible(esoc_udev->req_wait,
 					!kfifo_is_empty(&esoc_udev->req_fifo));
 		if (!err) {
-- 
1.9.1

